desc/io/input_reader.py

```diff
@@ -776,6 +778,185 @@ def write_desc_input(filename, inputs, header=""):  # noqa: C901 - fxn too compl
+            "M_grid": "M_grid",
+            "N_grid": "N_grid",
+        }.items():
+            f.write(key + " = {}\n".format(" " + str(eval("eq0.{0}".format(val)))))
+
+        print("DESC will automatically choose continuation parameters..\n")
+
+        f.write("\n\n# solver tolerances\n")
+        try:
+            f.write("ftol = {}\n".format(eq0._bdry_ratio))
+        except AttributeError:
+            print("ftol not given in the output. Setting to 1E-2...")
+            f.write("ftol = {}\n".format(float(1e-2)))
+
+        try:
+            f.write("xtol = {}\n".format(eq0._pres_ratio))
+        except AttributeError:
+            print("xtol not given in the output. Setting to 1E-6...")
+            f.write("xtol = {}\n".format(float(1e-6)))
+
+        try:
+            f.write("gtol = {}\n".format(eq0._curr_ratio))
+        except AttributeError:
+            print("gtol not given in the output. Setting to 1E-6...")
+            f.write("gtol = {}\n".format(float(1e-6)))
+
+        try:
+            f.write("maxiter = {}\n".format(eq0._pert_order))
+        except AttributeError:
+            print("maxiter not given in the output. Setting to 100...")
+            f.write("maxiter = {}\n".format(int(100)))
+
+        f.write("\n\n# solver methods\n")
+        try:
+            f.write("optimizer = {}\n".format(eq0._optimizer))
+        except AttributeError:
+            print("Optimizer absent in the output. Setting to lsq-exact...")
+            f.write("optimizer = {}\n".format("lsq-exact"))
+
+        try:
+            f.write("objective = {}\n".format(eq0._objective))
+        except AttributeError:
+            print("Objective not given in the output. Setting to force...")
+            f.write("objective = {}\n".format("force"))
+
+        f.write("spectral_indexing = {}\n".format(eq0._spectral_indexing))
+        f.write("node_pattern = {}\n".format(eq0._node_pattern))
+
+        f.write("\n# pressure and rotational transform/current profiles\n")
+
+        # can't import other stuff in io due to circular imports, so have to check by name
+        assert (
+            eq0.pressure.__class__.__name__ == "PowerSeriesProfile"
+        ), "Equilibrium must have power series profiles for ascii io"
+
+        pres_profile1 = eq0._pressure.params.tolist()
+
+        try:
+            iota_profile1 = eq0._iota.params.tolist()
+            assert (
+                eq0.iota.__class__.__name__ == "PowerSeriesProfile"
+            ), "Equilibrium must have power series profiles for ascii io"
+            char = "i"
+            if len(pres_profile1) >= len(iota_profile1):
+                pres_profile = pres_profile1
+
+                iota_profile = np.zeros((len(pres_profile),))
+                iota_profile[: len(iota_profile1)] = iota_profile1
+            else:
+                pres_profile1 = np.zeros((len(iota_profile1),))
+                pres_profile[: len(pres_profile)] = pres_profile
+        except AttributeError:
```

Comment:
probably a cleaner way to do this is to use an if statement and check `if eq.iota` for the first block, as `eq.iota= None` when it is a current constrained equilibrium

Reply:
Done.